compiler/src/org.graalvm.compiler.core.aarch64/src/org/graalvm/compiler/core/aarch64/AArch64NodeMatchRules.java

```diff
@@ -521,6 +522,26 @@ public ComplexMatchResult logicShift(BinaryNode binary, ValueNode a, BinaryNode
         return emitBinaryShift(op, a, shift);
     }
 
+    @MatchRule("(Negate (UnsignedRightShift=shift a Constant=b))")
+    @MatchRule("(Negate (RightShift=shift a Constant=b))")
+    @MatchRule("(Negate (LeftShift=shift a Constant=b))")
+    public ComplexMatchResult negShift(BinaryNode shift, ValueNode a, ConstantNode b) {
+        assert b.getStackKind().isNumericInteger();
+        assert a.getStackKind().isNumericInteger();
+        int maxAllowed = shift.getStackKind().getBitCount() - 1;
+        int shiftAmt = b.asJavaConstant().asInt();
+        if (shiftAmt > maxAllowed || shiftAmt < 0) {
+            return null;
+        }
```

Comment:
By clamping the value within the macroassembler, there shouldn't be the need for the above range check and bailout

Reply:
This change has been implemented.